addon/validations/factory.js

```diff
@@ -445,8 +445,8 @@ function getCPDependentKeysFor(attribute, validations, owner) {
     const cpDependents = Object.keys(options).reduce((arr, key) => {
       let option = options[key];
 
-      if(typeof option === 'object' && get(option, 'isDescriptor')) {
-        return arr.concat(get(option, '_dependentKeys').map(d => {
+      if(get(option, 'isDescriptor')) {
+        return arr.concat(getWithDefault(option, '_dependentKeys', []).map(d => {
```

Comment:
No need for getWithDefault here. Something like

`(get(option, '_dependentKeys') || []).map` is fine since this is not an ember object